tests/integ/modin/frame/test_apply.py

```diff
@@ -124,6 +143,17 @@ def foo(row) -> str:
         eval_snowpark_pandas_result(snow_df, df, lambda x: x.apply(foo, axis=1))
 
 
+@pytest.mark.xfail(strict=True, raises=NotImplementedError)
```

Comment:
I thought for NotImplementedError, we would typically use `pytest.raises(NotImplementedError)` instead of xfails, as we generally wanted to lower the number of xfails in our test suite?

Reply:
Yeah, the points you make do make sense. I think one benefit of `pytest.raises` is that you can also match on the error message. Also, looking at how commonly used each pattern is in our test suite (including Snowpark python tests), I could find about 300 instances of `xfail` vs about 1700 instances of `pytest.raises`. But still it's fair to say that both patterns are common enough.